### DISCUSSION TITLE
Multi-entités

### DISCUSSION THREAD

**Discussion by Nebrium** [dbarzin/mercator#335] [Q&A]
Bonjour,
J'ai une problématique dans l'utilisation de mercator. Je souhaite gérer plusieurs entités (comme un GHT par exemple), dans une seule base mercator.
Pour certaines choses je vais pouvoir m'adapter comme la partie infrastructure logique, ou la partie physique )je peux travailler par "site" comme étant une entité du GHT). Mais sur le système d'information, par exemple, je ne peux pas attribuer un processus ou macro processus par entité.
Idem quand je sors un rapport, il n'est pas possible de mettre des filtres pour sortir telle ou telle entité.
Est-ce qu'il y a une solution à mon problème ? ;-) Quelqu'un a déjà utilisé mercator en mode multientité ?
Merci pour vos retours.

**Comment by dbarzin** [comment]
Bonjour,
Pour le moment, il n'est pas possible de gérer ce cas d'utilisation. Cela nécessite une refonte complète de la base de données.
Par contre, il est possible de faire tourner plusieurs instance de Mercator sur un même serveur qui utilisent des bases de données différentes.
Cordialement,
Didier

**Comment by Nebrium** [comment]
Merci pour le retour. Je ne suis pas assez doué pour faire tourner plusieurs instances et plusieurs bases😅 Je vais tenter de faire quelque chose. 

Ce qui serait bien également c'est pouvoir extraire la page où on se trouve au format image ou pdf, est-ce que c'est prévu ? Par exemple dans Exploration, pouvoir construire quelque chose et l'extraire en PNG pour le réintégrer dans d'autres documents.

**Comment by dbarzin** [comment]
C'est assez simple. 
- Copier le répertoire Mercator vers Mercator2
- Éditer le fichier .env dans Mercator2 changer DB_DATABASE=mercator en DB_DATABASE=mercator2
- Puis continuer [l'installation](https://github.com/dbarzin/mercator/blob/master/INSTALL.fr.md) à partir du chapitre "Créer la base de données" 
- Au moment de la configuration Apache, il y a plusieurs possibilités. Le plus simple est d'utiliser un autre port comme le 81. Créer une mercator2.conf et remplacer :

     <VirtualHost *:80> par <VirtualHost *:81>
     DocumentRoot /var/www/mercator/public par DocumentRoot /var/www/mercator2/public

Puis faire :

    sudo a2ensite mercator2.conf

Enfin, redémarrez le service Apache pour activer les modifications :

sudo systemctl restart apache2

Pour l’extraction d'une image en PNG, c'est dans la feuille de route. Pour le moment tu peux toujours faire une capture d'écran :-)

**Comment by Nebrium** [comment]
Merci pour ton retour. J'avais déjà trouvé l'astuce ce matin pour faire plusieurs instances 😉 Le plus long a été de trouver où était la variable pour personnaliser le titre (panel.php) pour pas confondre mes instances 😂 J'ai lancé mes 5 instances, merci !

**Comment by dbarzin** [comment]
Tu risques d'avoir un problème lors de la mise à jour de l'application car tu as modifié un fichier provenant de git.
Je vais mettre le contenu de la variable APP_NAME qui se trouve dans le .env à la place.
Comme cela pas de problème lors de la mise à jour de l'application.

**Reply by Nebrium** [reply]
Oui je me doutais que lors des mises à jour ça allait écraser le fichier. Merci pour la modification 😉

**Comment by dbarzin** [comment]
C'est en place.
